File: CS/Standalone ExpressionEditor/UnboundExpressionPanel.cs
using DevExpress.Data;
using DevExpress.Data.ExpressionEditor;
using DevExpress.DataAccess.UI.ExpressionEditor;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ExpressionEditor;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Internal;
using DevExpress.XtraTreeList.Columns;
using System;
using System.ComponentModel;
using System.Linq;

namespace Standalone_ExpressionEditor
{
    [ToolboxItem(true)]
    [DesignerCategory("")]
    public class UnboundExpressionPanel : PanelControl
    {
        ExpressionEditorView view;
        object column;
        IDataColumnInfo columnContext;
        public UnboundExpressionPanel() : base() {
            BorderStyle = DevExpress.XtraEditors.Controls.BorderStyles.NoBorder;
        }
        void DestroyExpressionControls() {
            if (view != null) {
                Controls.Remove(view);
                view.Dispose();
            }
        }
        ExpressionEditorView CreateExpressionControl() {
            ExpressionEditorControl control = new ExpressionEditorControl();
            if (column is GridColumn)
                columnContext = new GridColumnIDataColumnInfoWrapper(column as GridColumn, GridColumnIDataColumnInfoWrapperEnum.ExpressionEditor);
            if (column is TreeListColumn)
                columnContext = column as IDataColumnInfo;
            control.Context = ExpressionEditorHelper.GetExpressionEditorContext(LookAndFeel, columnContext);
            ExpressionEditorView expressionView = new ExpressionEditorView(control.LookAndFeel, control);
            expressionView.Dock = System.Windows.Forms.DockStyle.Fill;
            expressionView.TopLevel = false;
            expressionView.FormBorderStyle = System.Windows.Forms.FormBorderStyle.None;
            expressionView.Ok += ExpressionView_Ok;
            expressionView.Cancel += ExpressionView_Cancel;
            expressionView.Visible = true;
            return expressionView;
        }
        public void StartEdit(object columnObject) {
            SuspendLayout();
            column = columnObject;
            DestroyExpressionControls();
            view = CreateExpressionControl();
            Controls.Add(view);
            string expression = string.Empty;
            if (column is GridColumn)
                expression = (column as GridColumn).UnboundExpression;
            if (column is TreeListColumn)
                expression = (column as TreeListColumn).UnboundExpression;
            view.ExpressionString = UnboundExpressionConvertHelper.ConvertToCaption(columnContext, expression);
            ResumeLayout();
        }

        private void ExpressionView_Ok(object sender, EventArgs e) {
            string expression = UnboundExpressionConvertHelper.ConvertToFields(columnContext, view.ExpressionString);
            if (column is GridColumn)
                (column as GridColumn).UnboundExpression = expression;
            if (column is TreeListColumn)
                (column as TreeListColumn).UnboundExpression = expression;
        }
        private void ExpressionView_Cancel(object sender, EventArgs e) {
            // TO DO
        }
    }
}

File: VB/Standalone ExpressionEditor/UnboundExpressionPanel.vb
Imports DevExpress.Data
Imports DevExpress.Data.ExpressionEditor
Imports DevExpress.DataAccess.UI.ExpressionEditor
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ExpressionEditor
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Internal
Imports DevExpress.XtraTreeList.Columns
Imports System
Imports System.ComponentModel
Imports System.Linq

Namespace Standalone_ExpressionEditor
	<ToolboxItem(True), DesignerCategory("")>
	Public Class UnboundExpressionPanel
		Inherits PanelControl

		Private view As ExpressionEditorView
		Private column As Object
		Private columnContext As IDataColumnInfo
		Public Sub New()
			MyBase.New()
			BorderStyle = DevExpress.XtraEditors.Controls.BorderStyles.NoBorder
		End Sub
		Private Sub DestroyExpressionControls()
			If view IsNot Nothing Then
				Controls.Remove(view)
				view.Dispose()
			End If
		End Sub
		Private Function CreateExpressionControl() As ExpressionEditorView
			Dim control As New ExpressionEditorControl()
			If TypeOf column Is GridColumn Then
				columnContext = New GridColumnIDataColumnInfoWrapper(TryCast(column, GridColumn), GridColumnIDataColumnInfoWrapperEnum.ExpressionEditor)
			End If
			If TypeOf column Is TreeListColumn Then
				columnContext = TryCast(column, IDataColumnInfo)
			End If
			control.Context = ExpressionEditorHelper.GetExpressionEditorContext(LookAndFeel, columnContext)
			Dim expressionView As New ExpressionEditorView(control.LookAndFeel, control)
			expressionView.Dock = System.Windows.Forms.DockStyle.Fill
			expressionView.TopLevel = False
			expressionView.FormBorderStyle = System.Windows.Forms.FormBorderStyle.None
			AddHandler expressionView.Ok, AddressOf ExpressionView_Ok
			AddHandler expressionView.Cancel, AddressOf ExpressionView_Cancel
			expressionView.Visible = True
			Return expressionView
		End Function
		Public Sub StartEdit(ByVal columnObject As Object)
			SuspendLayout()
			column = columnObject
			DestroyExpressionControls()
			view = CreateExpressionControl()
			Controls.Add(view)
			Dim expression As String = String.Empty
			If TypeOf column Is GridColumn Then
				expression = (TryCast(column, GridColumn)).UnboundExpression
			End If
			If TypeOf column Is TreeListColumn Then
				expression = (TryCast(column, TreeListColumn)).UnboundExpression
			End If
			view.ExpressionString = UnboundExpressionConvertHelper.ConvertToCaption(columnContext, expression)
			ResumeLayout()
		End Sub

		Private Sub ExpressionView_Ok(ByVal sender As Object, ByVal e As EventArgs)
			Dim expression As String = UnboundExpressionConvertHelper.ConvertToFields(columnContext, view.ExpressionString)
			If TypeOf column Is GridColumn Then
				TryCast(column, GridColumn).UnboundExpression = expression
			End If
			If TypeOf column Is TreeListColumn Then
				TryCast(column, TreeListColumn).UnboundExpression = expression
			End If
		End Sub
		Private Sub ExpressionView_Cancel(ByVal sender As Object, ByVal e As EventArgs)
			' TO DO
		End Sub
	End Class
End Namespace
